pinot-core/src/main/java/org/apache/pinot/core/data/manager/offline/DimensionTableDataManager.java

```diff
@@ -110,11 +127,19 @@ protected void doInit() {
     }
 
     if (_disablePreload) {
+      Object2LongOpenCustomHashMap<Object[]> lookupTable = new Object2LongOpenCustomHashMap<>(HASH_STRATEGY);
```

Comment:
The open map is very slow when rehashing because it re-compute hash code for all values. Java `HashMap` caches hash code instead. Can you try benchmarking it with more values and see if the performance gain still exists?

Reply:
This shouldn't be a concern here because lookup table :
- is pre-sized correctly before eager load 
- is not modified after creation
- is never loaded partially but from scratch in `DimensionTableDataManager.loadLookupTable()` .